comp110-worksheet-6/OxoBoard.cs

```diff
@@ -10,49 +10,316 @@ public enum Mark { None, O, X };
+                            if (count == inARow)
+                            {
+                                return Mark.X;
+                            }
+                            y_check++;
+                        }
+
+                        else break;
+                    }
+
+                }
+            }
+
+            //Checks rows
+
+            for (int y = 0; y < height; y++)
+            {
+                for (int x = 0; x < width; x++)
+                {
+                    int x_check = x;
+                    for (count = 1; count <= inARow; count++)
+                    {
+                        if (x_check >= width)
+                        { break; }
+                        if (Game_Board[x_check, y] == Mark.X)
+                        {
+                            if (count == inARow)
+                            {
+                                return Mark.X;
+                            }
+                            x_check++;
+                        }
+
+                        else break;
+                    }
+
+
+                }
+            }
+
+
+            //Checks for diagonals
+            //Checking for x+ and y+
+
+            for (int x = 0; x < width; x++)
+            {
+                for (int y = 0; y < height; y++)
+                {
+                    int x_check = x;
+                    int y_check = y;
+                    for (count = 1; count <= inARow; count++)
+                    {
+                        if (x_check >= width || y_check >= height)
+                        { break; }
+                        if (Game_Board[x_check, y_check] == Mark.X)
+                        {
+                            if (count == inARow)
+                            {
+                                return Mark.X;
+                            }
+                            x_check++;
+                            y_check++;
+                        }
+
+                        else break;
+                    }
+
+                }
+            }
+
+            //Checking for x+ and y-
+
+            for (int x = 0; x < width; x++)
+            {
+                for (int y = 0; y < height; y++)
+                {
+                    int x_check = x;
+                    int y_check = y;
+                    for (count = 1; count <= inARow; count++)
+                    {
+                        if (x_check >= width || y_check >= height || y_check < 0)
+                        { break; }
+                        if (Game_Board[x_check, y_check] == Mark.X)
+                        {
+                            if (count == inARow)
+                            {
+                                return Mark.X;
+                            }
+                            x_check++;
+                            y_check--;
+                        }
+
+                        else break;
+                    }
+                }
+            }
+
+            //check for O
```

Comment:
The checks for O are almost identical to the checks for X -- these could have been combined into a loop or a helper function to cut down on code duplication